File: HotDesk_PRAI/App_Module/GridFooter.ascx.vb

Partial Class UserControl_GridFooter
    Inherits System.Web.UI.UserControl

    Private setting As Control.Base
    Private logSetting As Control.LogBase

    Protected Sub Page_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
        If Not IsPostBack Then
            Me.BindPage()
            Me.setNavigator()
        End If
        'If Not IsPostBack Then
        'End If
    End Sub

    Private _audit As Boolean = False
    Public WriteOnly Property Audit() As Boolean
        Set(ByVal value As Boolean)
            _audit = value
        End Set
    End Property

    Private _total As Integer = 0
    Public Property TotalRecords() As Integer
        Get
            Return _total
        End Get
        Set(ByVal Value As Integer)
            _total = Value
            Me._totalPage = Math.Ceiling(_total / Library.Root.Other.BusinessLogicBase.MaxQuantityPerPage)
        End Set
    End Property

    Private _totalPage As Integer = 0
    Public ReadOnly Property TotalPage() As Integer
        Get
            Return Math.Ceiling(_total / Library.Root.Other.BusinessLogicBase.MaxQuantityPerPage)
        End Get
    End Property

    Protected Sub ddlPage_SelectedIndexChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles ddlPage.SelectedIndexChanged
        If _audit Then
            logSetting = CType(Me.Page, Control.LogBase)
            logSetting.PageNo = ddlPage.SelectedItem.Text
            Response.Redirect(logSetting.GenerateList)
        Else
            setting = CType(Me.Page, Control.Base)
            setting.PageNo = ddlPage.SelectedItem.Text
            Response.Redirect(setting.GenerateList)
        End If
    End Sub

    Protected Sub btnNext_Click(ByVal sender As Object, ByVal e As System.Web.UI.ImageClickEventArgs) Handles btnNext.Click
        If _audit Then
            logSetting = CType(Me.Page, Control.LogBase)

            If ddlPage.Items.Count > 0 Then
                If (logSetting.PageNo >= ddlPage.Items(ddlPage.Items.Count - 1).Value) Then
                    Return
                End If
            Else
                Return
            End If

            logSetting.PageNo = logSetting.PageNo + 1
            Response.Redirect(logSetting.GenerateList)
        Else
            setting = CType(Me.Page, Control.Base)

            If ddlPage.Items.Count > 0 Then
                If (setting.PageNo >= ddlPage.Items(ddlPage.Items.Count - 1).Value) Then
                    Return
                End If
            Else
                Return
            End If

            setting.PageNo = setting.PageNo + 1
            Response.Redirect(setting.GenerateList)
        End If
    End Sub

    Protected Sub btnPrevious_Click(ByVal sender As Object, ByVal e As System.Web.UI.ImageClickEventArgs) Handles btnPrevious.Click
        If _audit Then
            logSetting = CType(Me.Page, Control.LogBase)
            If (logSetting.PageNo <= 0) Then
                Return
            End If

            logSetting.PageNo = logSetting.PageNo - 1
            Response.Redirect(logSetting.GenerateList)
        Else
            setting = CType(Me.Page, Control.Base)
            If (setting.PageNo <= 0) Then
                Return
            End If

            setting.PageNo = setting.PageNo - 1
            Response.Redirect(setting.GenerateList)
        End If
    End Sub

    Protected Sub btnLast_Click(ByVal sender As Object, ByVal e As System.Web.UI.ImageClickEventArgs) Handles btnLast.Click
        If _audit Then
            logSetting = CType(Me.Page, Control.LogBase)
            logSetting.PageNo = ddlPage.Items(ddlPage.Items.Count - 1).Value
            Response.Redirect(logSetting.GenerateList)
        Else
            setting = CType(Me.Page, Control.Base)
            setting.PageNo = ddlPage.Items(ddlPage.Items.Count - 1).Value
            Response.Redirect(setting.GenerateList)
        End If
    End Sub

    Protected Sub btnFirst_Click(ByVal sender As Object, ByVal e As System.Web.UI.ImageClickEventArgs) Handles btnFirst.Click
        If _audit Then
            logSetting = CType(Me.Page, Control.LogBase)
            logSetting.PageNo = 1
            Response.Redirect(logSetting.GenerateList)
        Else
            setting = CType(Me.Page, Control.Base)
            setting.PageNo = 1
            Response.Redirect(setting.GenerateList)
        End If
    End Sub

    Private Sub BindPage()
        If _audit Then
            logSetting = CType(Me.Page, Control.LogBase)
            If logSetting.PageNo > Me._totalPage Then
                logSetting.PageNo = Me._totalPage
            End If
        Else
            setting = CType(Me.Page, Control.Base)
            If setting.PageNo > Me._totalPage Then
                setting.PageNo = Me._totalPage
            End If
        End If

        ddlPage.Items.Clear()
        For i As Integer = 1 To Me._totalPage
            ddlPage.Items.Add(i)
        Next

        lblTotalRecord.Text = String.Format("Total of {0} records found", Me.TotalRecords)
    End Sub

    Protected Sub setNavigator()
        If _audit Then
            logSetting = CType(Me.Page, Control.LogBase)
        Else
            setting = CType(Me.Page, Control.Base)
        End If


        If (Me._total <= 1) Then
            Me.btnFirst.Visible = False
            Me.btnLast.Visible = False
            Me.btnNext.Visible = False
            Me.btnPrevious.Visible = False
            Return
        Else
            Me.btnFirst.Visible = True
            Me.btnLast.Visible = True
            Me.btnNext.Visible = True
            Me.btnPrevious.Visible = True
        End If

        Me.ddlPage.SelectedIndex = If(_audit, logSetting.PageNo - 1, setting.PageNo - 1)
        If ddlPage.SelectedIndex = 0 Then
            Me.btnFirst.Visible = False
            Me.btnPrevious.Visible = False
        Else
            Me.btnFirst.Visible = True
            Me.btnPrevious.Visible = True
        End If

        Dim iPage As Integer = Me.TotalPage
        If ddlPage.SelectedIndex >= iPage - 1 Then
            Me.btnLast.Visible = False
            Me.btnNext.Visible = False
        Else
            Me.btnLast.Visible = True
            Me.btnNext.Visible = True
        End If
    End Sub
End Class

File: TMS_HotDesk/App_Module/GridFooter.ascx.cs
using System;

public partial class UserControl_GridFooter : System.Web.UI.UserControl
{
    private Control.Base setting;
    private Control.LogBase logSetting;

    protected void Page_Load(object sender, EventArgs e)
    {
        if (!IsPostBack)
        {
            this.BindPage();
            this.setNavigator();
        }
    }

    private bool _audit = false;
    public bool Audit { set { _audit = value; } }

    private int _total = 0;
    public int TotalRecords
    {
        get { return _total; }
        set
        {
            _total = value;
            this._totalPage = (int)Math.Ceiling((double)_total / Library.Root.Other.BusinessLogicBase.MaxQuantityPerPage);
        }
    }

    private int _totalPage = 0;
    public int TotalPage
    {
        get { return (int)Math.Ceiling((double)_total / Library.Root.Other.BusinessLogicBase.MaxQuantityPerPage); }
    }

    protected void ddlPage_SelectedIndexChanged(object sender, EventArgs e)
    {
        if (_audit)
        {
            logSetting = (Control.LogBase)this.Page;
            logSetting.PageNo = int.Parse(ddlPage.SelectedItem.Text);
            Response.Redirect(logSetting.GenerateList);
        }
        else
        {
            setting = (Control.Base)this.Page;
            setting.PageNo = int.Parse(ddlPage.SelectedItem.Text);
            Response.Redirect(setting.GenerateList);
        }
    }

    protected void btnNext_Click(object sender, System.Web.UI.ImageClickEventArgs e)
    {
        if (_audit)
        {
            logSetting = (Control.LogBase)this.Page;
            if (ddlPage.Items.Count > 0)
            {
                if (logSetting.PageNo >= int.Parse(ddlPage.Items[ddlPage.Items.Count - 1].Value)) return;
            }
            else return;
            logSetting.PageNo = logSetting.PageNo + 1;
            Response.Redirect(logSetting.GenerateList);
        }
        else
        {
            setting = (Control.Base)this.Page;
            if (ddlPage.Items.Count > 0)
            {
                if (setting.PageNo >= int.Parse(ddlPage.Items[ddlPage.Items.Count - 1].Value)) return;
            }
            else return;
            setting.PageNo = setting.PageNo + 1;
            Response.Redirect(setting.GenerateList);
        }
    }

    protected void btnPrevious_Click(object sender, System.Web.UI.ImageClickEventArgs e)
    {
        if (_audit)
        {
            logSetting = (Control.LogBase)this.Page;
            if (logSetting.PageNo <= 0) return;
            logSetting.PageNo = logSetting.PageNo - 1;
            Response.Redirect(logSetting.GenerateList);
        }
        else
        {
            setting = (Control.Base)this.Page;
            if (setting.PageNo <= 0) return;
            setting.PageNo = setting.PageNo - 1;
            Response.Redirect(setting.GenerateList);
        }
    }

    protected void btnLast_Click(object sender, System.Web.UI.ImageClickEventArgs e)
    {
        if (_audit)
        {
            logSetting = (Control.LogBase)this.Page;
            logSetting.PageNo = int.Parse(ddlPage.Items[ddlPage.Items.Count - 1].Value);
            Response.Redirect(logSetting.GenerateList);
        }
        else
        {
            setting = (Control.Base)this.Page;
            setting.PageNo = int.Parse(ddlPage.Items[ddlPage.Items.Count - 1].Value);
            Response.Redirect(setting.GenerateList);
        }
    }

    protected void btnFirst_Click(object sender, System.Web.UI.ImageClickEventArgs e)
    {
        if (_audit)
        {
            logSetting = (Control.LogBase)this.Page;
            logSetting.PageNo = 1;
            Response.Redirect(logSetting.GenerateList);
        }
        else
        {
            setting = (Control.Base)this.Page;
            setting.PageNo = 1;
            Response.Redirect(setting.GenerateList);
        }
    }

    private void BindPage()
    {
        if (_audit)
        {
            logSetting = (Control.LogBase)this.Page;
            if (logSetting.PageNo > this._totalPage) logSetting.PageNo = this._totalPage;
        }
        else
        {
            setting = (Control.Base)this.Page;
            if (setting.PageNo > this._totalPage) setting.PageNo = this._totalPage;
        }

        ddlPage.Items.Clear();
        for (int i = 1; i <= this._totalPage; i++)
            ddlPage.Items.Add(i.ToString());

        lblTotalRecord.Text = string.Format("Total of {0} records found", this.TotalRecords);
    }

    protected void setNavigator()
    {
        if (_audit)
            logSetting = (Control.LogBase)this.Page;
        else
            setting = (Control.Base)this.Page;

        if (this._total <= 1)
        {
            this.btnFirst.Visible = false;
            this.btnLast.Visible = false;
            this.btnNext.Visible = false;
            this.btnPrevious.Visible = false;
            return;
        }
        else
        {
            this.btnFirst.Visible = true;
            this.btnLast.Visible = true;
            this.btnNext.Visible = true;
            this.btnPrevious.Visible = true;
        }

        this.ddlPage.SelectedIndex = _audit ? logSetting.PageNo - 1 : setting.PageNo - 1;
        if (ddlPage.SelectedIndex == 0)
        {
            this.btnFirst.Visible = false;
            this.btnPrevious.Visible = false;
        }
        else
        {
            this.btnFirst.Visible = true;
            this.btnPrevious.Visible = true;
        }

        int iPage = this.TotalPage;
        if (ddlPage.SelectedIndex >= iPage - 1)
        {
            this.btnLast.Visible = false;
            this.btnNext.Visible = false;
        }
        else
        {
            this.btnLast.Visible = true;
            this.btnNext.Visible = true;
        }
    }
}
